### ISSUE TITLE
Returning http headers behave differently on each platform

### ISSUE THREAD

**Issue by snpori** [capacitor-community/http#32]
**Describe the bug**
I'm trying to read a custom http header which is set a server.
I've noticed that on each platform. ios/android/browser the headers are structured diffently which requires a diffrent handling on each platform. 
This leads to the following:
```
 // if web
 token = res.headers['some-token']; // chrome on desktop is returning all lowercase header names
 //else if iOS
  token = res.headers['Some-Token']; // Case sensitive on iOS
 // else if Android
 token = res.headers[1]['Some-Token']; // Android is returning a 2d headers array 
```
                    
**Expected behavior**
Reading headers should be consistent on all platforms

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Android10, Ios13
 - Browser Chrome
 - Version  0.2.1

**Additional context**
This is tested in ionic4 project

**Comment by thomasvidas** [comment]
Hello, could you provide a reproducible example project?

**Comment by thomasvidas** [comment]
Is it only for custom headers or is it for all headers?

**Comment by Landschaft** [comment]
I've observed the same problem. It exists for _all_ headers. In particular, the implementation on Android does not comply with the typescript structure definition:

    export interface HttpResponse {
      data: any;
      status: number;
      headers: HttpHeaders;
      url: string;
    }

The `headers` field is defined as type `HttpHeaders`, but the Android plugin implementation returns an **array** of `HttpHeaders`.

**Comment by Landschaft** [comment]
HTTP headers are _not_ case-sensitive, BTW.

**Comment by thomasvidas** [comment]
Fixed by #99
